Release 3.1.0-rc1

rbott opened the pull request:
This prepares the release of the 3.1.0-rc1 version of Ganeti. I tried to come up with a valid summary of the changes and include everything that is relevant to end users and/or distro maintainers.

Do you think anything is missing (or not accurate)?

dannyman reviewed (commented):
LGTM!

saschalucas commented:
thanks @rbott. It seams that `3.1` should have been in `ganeti.cabal` from the beginning. Can you please include this in your preparation?

saschalucas commented:
Maybe we can stress the point, that `--enable-regex-tdfa` is the only option from Debian Trixie onward, the only one tested in CI/QA **and** that `tdfa` does not provide PCRE type regexps. The place where this has an impact is unknown. Best guess seems to point to `gnt-filter(8)`.

rbott commented:
> Maybe we can stress the point, that `--enable-regex-tdfa` is the only option from Debian Trixie onward, the only one tested in CI/QA **and** that `tdfa` does not provide PCRE type regexps. The place where this has an impact is unknown. Best guess seems to point to `gnt-filter(8)`.

I have updated the NEWS file accordingly and also added a stance about `extstorage` (which might break if people use a custom storage provider that is somehow incompatible with the `-blockdev` parameter). Only situation I can think of would be a provider that does return a userspace URL instead of a blockdevice (if that even exists).

saschalucas commented on NEWS:
```diff
@@ -2,6 +2,166 @@ News
 ====
 
 
+Version 3.1.0 rc1
```
Just for completeness, I assume skipping a beta release is intentional here? If yes, I'm fine with no beta.

saschalucas commented on NEWS:
```diff
@@ -2,6 +2,166 @@ News
+Upgrading is supported from 3.0.x versions.
+
+If you are using the KVM hypervisor, you should consider setting the parameter
+`machine_version` to `pc`. If it is unset, Ganeti will auto-detect the machine
```
I generally would not recommend setting `machine_version` to `pc`. Instead the admin should chose an explicit machine version (i.e. `pc-i440fx-7.2`). This guaranties safe live migration across qemu upgrades!
Only for the absentminded admin Ganeti defaults to `pc` on new clusters, which represents the _current_ qemu version and is not stable for live migration between different qemu versions. It just prevents a possible breaking switch to `q35`in future qemu versions.

rbott commented:
After some discussion with @saschalucas we decided to revert the changes regarding `machine_version` for 3.1

This has been reflected in the NEWS file and also in #1827

saschalucas reviewed (approved):
Thanks @rbott. LGTM.

apoikos reviewed (approved):
LGTM, thank you all for making this release possible!

apoikos commented:
I will perform a couple of tests before actually tagging the release, please bear with me for a little bit :-)
